src/main/kotlin/com/michaelsgroi/baseballreference/Career.kt

```diff
@@ -18,8 +24,51 @@ data class Career(
     val playerName: String,
     val war: Double,
     val seasonLines: List<SeasonLine>,
+    val awardVoteLines: List<RankedAwardShare>,
     val warPercentile: Double = 0.0,
 ) {
+    private val seasonLinesBySeason: Map<Int, List<SeasonLine>> by lazy { seasonLines.groupBy { it.season() } }
+    private val awardVotesBySeason: Map<Int, List<RankedAwardShare>> by lazy { awardVoteLines.groupBy { it.season } }
+
+    init {
+        validate()
+    }
+
+    private fun validate() {
+        // player id's
+        val seasonLinePlayerIds = seasonLines.map { it.playerId() }.toSet()
+        val awardLinePlayerIds = awardVoteLines.map { it.playerId }.toSet()
+        require(
+            seasonLinePlayerIds.size == 1
+                    && (awardLinePlayerIds.isEmpty() || awardLinePlayerIds.size == 1)
+                    && seasonLinePlayerIds.first() == playerId
+                    && (awardLinePlayerIds.isEmpty() || awardLinePlayerIds.first() == playerId)
+        ) {
+            "seasonLinePlayerIds($seasonLinePlayerIds) and " +
+                    "awardLinePlayerIds($awardLinePlayerIds) must match parent playerId=$playerId"
+        }
+
+        // seasons
+        val seasonLineSeasons = seasonLinesBySeason.keys
+        val awardVoteSeasons = awardVotesBySeason.keys
+        require(seasonLineSeasons.containsAll(awardVoteSeasons))
+        { "award vote season($awardVoteSeasons) contains a season not found in season line seasons($seasonLineSeasons)" }
+
+        // leagues
+        val seasonLineLeagues = seasonLines.map { it.league() }.toSet()
+        val awardLineLeagues = awardVoteLines.map { it.league }.toSet().minus("ML")
+        require(seasonLineLeagues intersect awardLineLeagues == awardLineLeagues)
+        { "award vote league($seasonLineLeagues) contains a league not found in season line leagues($seasonLineLeagues)" }
+
+        // awards
+        awardVotesBySeason.values.forEach { awardLines ->
+            val awardNames = awardLines.map { awardLine -> awardLine.awardName }
+            require(awardNames.size == awardNames.toSet().size) {
+                "duplicate awards found: $awardNames"
```

Comment:
"found" -> "found in" (this isn't listing the duplicate awards—it lists _all_ the awards)

Reply:
fixed ... bfd90fbdad00ddf8b9a346e5bab6f9782ca8311f